src/proxy-worker.ts

```diff
@@ -91,7 +91,7 @@ class Tunnel extends nodeTunnel.Tunnel {
 				return socket;
 			} else {
 				const vpnHost = await device.getDeviceVpnHost(uuid, auth);
-				if (vpnHost.id === this.serviceId) {
+				if (vpnHost.id !== this.serviceId) {
```

Comment:
I fear this will break everything @fisehara . By everything, I mean to say that all devices will get a `500`. On line 80, we check if the device is served by the VPN instance running this code. In that case `vpnHost.id === this.serviceId` will be true by design. In the `else` clause, `vpnHost.id` should never be equal to `this.serviceId`. If they are equal (which is the original check on line 94), then it means the system is not consistent which is why we return a `500` in that case.
The fact that such a check was introduced implies that this case did arise (or was envisioned by whoever wrote the code). @Page- , do you happen to remember/know why this was done? It seems that our mechanism of relaying device connectivity/disconnectivity can fail to happen (or get delayed). Pods getting recycled is one scenario that comes to mind.